frontend/src/components/inputs/DateTime.vue

```diff
@@ -1,7 +1,9 @@
 <template>
   <BaseInput :label="label" :args="args" :vertical="vertical" :required="required">
     <template v-slot:field>
-      <input class="form-control" type="datetime-local" :value="modelValue" @input="onInput"
+      <input class="form-control" type="datetime-local"
+             :value="modelValue && convertToDateTimeLocalString(modelValue)"
```

Comment:
This feels wrong for. I think we shouldn't modify the value in here like this, but rather take care of the modelValue in the script section - it'll be more intuitive/readable. The main problem I have with this component is that I don't know what the model value type is supposed to be. Is  it `Date`? String?

In either case I'd rather force the model value to always be Date and than handle the **string -> Date** conversion in the `onInput`, which will self-correct the model value to be again of type Date... simply:

```
onInput(event) {
  const resolvedDate = new Date(event.target.value);
  this.$emit('update:modelValue', resolvedDate);
}
```

...because datetime input can handle the **Date -> string** conversion internally on its own, right?

P.S. I hope that

Reply:
Everything works correctly on input. The problem occurs when there is a value coming from the backend and it's formatted like "2024-04-08T15:15:00+03:00" which the datetime-local input does not like and needs to be reformatted in order to get displayed.

> The main problem I have with this component is that I don't know what the model value type is supposed to be. Is it Date? String?

String

> ...because datetime input can handle the Date -> string conversion internally on its own, right?

Actually, it only accepts string as value :/

> This feels wrong. I think we shouldn't modify the value in here like this, but rather take care of the modelValue in the script section

Yes, you got a point, I agree.

Reply:
huh.. I tried this out and it worked fine.. the thing is that when you pass the incoming string to `new Date()` it parses without any issues and then you can safely pass the `Date` type to the input directly.

Reply:
weird, I can't recall what I did... I must have done something wrong. My apologies, it looks like the input can't indeed handle the Date type. So we can't make this any nicer.